packages/vscode-dependencies-validation/src/extension.ts

```diff
@@ -1,27 +1,96 @@
-import { Uri, ExtensionContext, workspace, window } from "vscode";
 import {
-  NPMDependencyIssue,
+  Uri,
+  ExtensionContext,
+  workspace,
+  window,
+  languages,
+  DiagnosticCollection,
+  OutputChannel,
+  commands,
+  CodeActionKind,
+} from "vscode";
+import {
+  NpmLsResult,
   findDependencyIssues,
 } from "@sap-devx/npm-dependencies-validation";
+import { NPMIssuesActionProvider } from "./npmIssuesActionProvider";
+import { fixAllDepIssuesCommand } from "./commands";
+import { FIX_ALL_ISSUES_COMMAND } from "./constants";
+import { refreshDiagnostics } from "./diagnostics";
+
+const PACKAGE_JSON = "package.json";
+const PACKAGE_JSON_PATTERN = `**​/${PACKAGE_JSON}`;
+const extName = "vscode-dependency-validation";
 
 export function activate(context: ExtensionContext) {
+  const outputChannel = window.createOutputChannel(extName);
+  const diagnosticCollection = createDiagnosticCollection(context, extName);
+
```

Comment:
- should we delay "findIssues" from startup performance considerations?
- does this scan all package.json files regardless if they are open or not? or only the open files?